### PR TITLE
add zero bucket if not present.

### PR DESCRIPTION
**Pull request by rghetia**

### CODE REVIEW

File: stats.go [hunk 1]
```diff
@@ -458,6 +459,27 @@ func newTypedValue(vd *view.View, r *view.Row) *monitoringpb.TypedValue {
 	return nil
 }
 
+func shouldInsertZeroBound(bounds ...float64) bool {
+	if len(bounds) > 0 && bounds[0] != 0.0 {
```
Comment (songy23) [inline stats.go]: Should we check if the first bound is negative?

Comment (mayurkale22) [reply stats.go]: I think no need to check for negative or zero bucket bounds, as we drop these bounds at top level API. I am not sure how it's done in Go, you may want to refer Java's implementation -> https://github.com/census-instrumentation/opencensus-java/blob/17e327f21921f981951608727602203bbc82238a/api/src/main/java/io/opencensus/stats/BucketBoundaries.java#L64-L90 and https://github.com/census-instrumentation/opencensus-java/blob/master/exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java#L153-L156

Comment (rghetia) [reply stats.go]: @songy23  negative value is checked [here](https://github.com/census-instrumentation/opencensus-go/blob/950a67f393d867cfbe91414063b69e511f42fefb/stats/view/view.go#L97-L101)

@mayurkale22 the reason for checking 0 value is that you could have different version (older version) of core library which would send 0 bound bucket. So you need to insert it conditionally.

File: metrics.go [hunk 2]
```diff
@@ -461,22 +461,28 @@ func protoToMetricPoint(value interface{}) (*monitoringpb.TypedValue, error) {
...
 				if ok && bexp != nil && bexp.Explicit != nil {
+					insertZeroBound = shouldInsertZeroBound(bexp.Explicit.Bounds...)
 					mv.DistributionValue.BucketOptions = &distributionpb.Distribution_BucketOptions{
 						Options: &distributionpb.Distribution_BucketOptions_ExplicitBuckets{
 							ExplicitBuckets: &distributionpb.Distribution_BucketOptions_Explicit{
-								Bounds: bexp.Explicit.Bounds[:],
+								// The first bucket bound should be 0.0 because the Metrics first bucket is
+								// [0, first_bound) but Stackdriver monitoring bucket bounds begin with -infinity
+								// (first bucket is (-infinity, 0))
+								Bounds: addZeroBound(insertZeroBound, bexp.Explicit.Bounds...),
```
Comment (songy23) [inline metrics.go]: Optional nit: for better readability consider
```go
if insertZeroBound {
  addZeroBound(bexp.Explicit.Bounds...)
}
```

File: metrics.go [hunk 3]
```diff
@@ -461,22 +461,28 @@ func protoToMetricPoint(value interface{}) (*monitoringpb.TypedValue, error) {
...
 							},
 						},
 					}
 				}
 			}
+			mv.DistributionValue.BucketCounts = addZeroBucketCount(insertZeroBound, bucketCounts(dv.Buckets)...)
```
Comment (songy23) [inline metrics.go]: Same here.

File: stats.go (hunk 1 above)
Comment (mayurkale22) [reply stats.go]: SGTM

File: metrics.go (hunk 2 above)
Comment (rghetia) [reply metrics.go]: @songy23 are you ok with the renaming the method or you want me to make the change as shown above?

Comment (songy23) [reply metrics.go]: Renaming the method sounds good.

**Comment by rghetia** [comment]
fixes #87

**Pull request merged**